src/parallel/impl_par_methods.rs

```diff
@@ -177,6 +177,48 @@ macro_rules! zip_impl {
                 self.par_map_assign_into(into, f)
             }
 
+            /// Parallel version of `fold`.
+            ///
+            /// Splits the producer in multiple tasks which each accumulate a single value
+            /// using the `fold` closure. Those tasks are executed in parallel and their results
+            /// are then combined to a single value using the `reduce` closure.
+            ///
+            /// The `identity` closure provides the initial values for each of the tasks and
+            /// for the final reduction.
+            ///
+            /// Note that the splitting of the producer into multiple tasks is _not_ deterministic
+            /// which needs to be considered when the accuracy of such an operation is analyzed.
+            ///
+            /// ## Examples
+            ///
+            /// ```rust
+            /// use ndarray::{Array, Zip};
+            ///
+            /// let a = Array::<usize, _>::ones((128, 1024));
+            /// let b = Array::<usize, _>::ones(128);
+            ///
+            /// let sum = Zip::from(a.rows()).and(b.view()).par_fold(
```

Comment:
```suggestion
            /// let weighted_sum = Zip::from(&a).and(&b).par_fold(
```

`weighted_sum` is just a suggestion.

Reply:
You could if you want sync up even more with the existing `Zip::fold` in terms of "presentation". By no means necessary, just saying what I'm thinking - (see existing ones to see that they are not perfect). If you want, we mention something like par for each's `This is a shorthand for using .into_par_iter().for_each() on Zip.`